packages/server/src/api_auth/index.ts

```diff
@@ -507,6 +512,15 @@ router.post('/logout', authenticateUser, async (req, res) => {
  *
  */
 router.post('/logout_all', authenticateUser, async (req, res) => {
+     try {
+        // Validate the request body using the Zod schema
+        logoutAllSchema.parse(req.locals);
+    } catch (error) {
+        if (error instanceof ZodError && !error.isEmpty) {
+            res.status(400).send({ error: "Token and email are required and must be non-empty" });
+            return;
+        }
+    }
     // res.locals should contain email and token
     if (!('token' in res.locals) || !('email' in res.locals)) {
```

Comment:
We can delete this block and all other old validation logic.

Reply:
Ok, will do it shortly